File: 347-top-k-frequent-elements/top-k-frequent-elements.py
class Solution(object):
    def topKFrequent(self, nums, k):
        """
        :type nums: List[int]
        :type k: int
        :rtype: List[int]
        """

        el_and_appearance = {}

        for i in nums:
            if i in el_and_appearance:
                el_and_appearance[i] += 1
            else:
                el_and_appearance[i] = 1

        return sorted(el_and_appearance, key=el_and_appearance.get, reverse=True)[:k]

File: 347-top-k-frequent-elements/top-k-frequent-elements.cpp
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
        unordered_map<int, int> appearanceCounter;
        vector<int> res(k);
        vector<set<int>> appearAsIndex(nums.size() + 1);
        int insertCounter = 0;
        for(int i : nums)
        {
            auto curr = appearanceCounter.find(i);
            if(curr == appearanceCounter.end())
            {
                appearanceCounter[i] = 1;
                appearAsIndex[1].insert(i);
            }
            else
            {
                appearAsIndex[curr->second].erase(i);
                curr->second++;
                appearAsIndex[curr->second].insert(i);
            }
        }
        for(int i = appearAsIndex.size() - 1; i >= 0 && insertCounter < k; i--)
        {
            for(int j : appearAsIndex[i])
            {
                res[insertCounter] = j;
                if(++insertCounter == k)
                    break;
            }
        }
        return res;
    }
};